addons/pos_self_order/static/tests/unit/utils.js

```diff
@@ -0,0 +1,57 @@
+import { uuidv4 } from "@point_of_sale/utils";
+import {
+    getService,
+    makeMockEnv,
+    onRpc,
+    patchWithCleanup,
+    MockServer,
+} from "@web/../tests/web_test_helpers";
+import { session } from "@web/session";
+import { registry } from "@web/core/registry";
+
+export function patchSession() {
+    // Do not change these variables, they are in accordance with the setup data
+    patchWithCleanup(session, {
+        data: {
+            config_id: 1,
+            self_ordering_mode: "kiosk",
+        },
+        db: "pos-self",
+    });
+}
+
+export function initMockRpc() {
+    onRpc(`/pos-self/relations/1`, () => MockServer.env["pos.config"].load_data_params());
+    onRpc(`/pos-self/data/1`, () => MockServer.env["pos.config"].load_self_data());
+}
+
+const removePosServices = (serviceNames) =>
+    serviceNames.forEach((serviceName) => registry.category("services").remove(serviceName));
+
+export const setupSelfPosEnv = async () => {
+    odoo.access_token = uuidv4();
+    odoo.info = {
+        isEnterprise: true,
+    };
+
+    // Removing `pos` and its dependent services to avoid conflicts during `self_order` data loading.
+    // Both `pos` and `self_order` rely on `pos_data`, but some models required by `self_order` (e.g., `res.users`)
+    // are missing when `pos` is loaded. Hence, these services are excluded.
+    removePosServices(["contextual_utils_service", "report", "pos"]);
```

Comment:
You can inline directly here


```suggestion
    const serviceNames = ["contextual_utils_service", "report", "pos"];
    serviceNames.forEach((serviceName) => registry.category("services").remove(serviceName));
```